src/routing_table/prefix.rs

```diff
@@ -24,25 +23,26 @@ use super::xorable::Xorable;
 
 /// A group prefix, i.e. a sequence of bits specifying the part of the network's name space
 /// consisting of all names that start with this sequence.
-#[derive(Clone, Copy, Default, Eq)]
+#[derive(Clone, Copy, Default, Eq, RustcDecodable, RustcEncodable)]
 pub struct Prefix<T: Clone + Copy + Default + Binary + Xorable> {
-    bit_count: usize,
+    bit_count: u16,
     name: T,
 }
 
 impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
-    /// Creates a new `Prefix` with the first `bit_count` bits of `name`.
-    /// Insignificant bits are all set to 0.
+    /// Creates a new `Prefix` with the first `bit_count` bits of `name`. Insignificant bits are all
+    /// set to 0. If `bit_count` exceeds the size of `T` in bits, then it is reduced to this lower
+    /// value.
     pub fn new(bit_count: usize, name: T) -> Prefix<T> {
         Prefix {
-            bit_count: bit_count,
+            bit_count: cmp::min(bit_count, T::bit_len()) as u16,
             name: name.set_remaining(bit_count, false),
         }
     }
 
     /// Returns `self` with an appended bit: `0` if `bit` is `false`, and `1` if `bit` is `true`.
     pub fn pushed(mut self, bit: bool) -> Prefix<T> {
-        self.name = self.name.with_bit(self.bit_count, bit);
+        self.name = self.name.with_bit(self.bit_count(), bit);
```

Comment:
I think it already returns the name unchanged.  I'm doing another change anyway, so I'll update the comment after checking that I'm right about the behaviour.